src/parse.js

```diff
@@ -338,13 +371,17 @@ function parse(source, root, options) {
 
                 case "required":
                 case "repeated":
+                    if (edition)
+                        throw illegal(token);
                     parseField(type, token);
                     break;
 
                 case "optional":
```

Comment:
Added tests for the grammar changes. As for the second point, I believe this is already covered by feature resolution tests since there are no quotes.

Reply:
I don't think that second case is covered, since you have no reserved fields.  What I'm talking about is that for a proto2 message:
```
message Foo {
  reserved "bar", "baz";
}
```

becomes this in edition 2023:
```
message Foo {
  reserved bar, baz;
}
```

Reply:
Added tests.